pkg/apis/rbac/types.go

```diff
@@ -0,0 +1,252 @@
+	Roles map[string]*Role
+}
+
+// PolicyBinding is a object that holds all the RoleBindings for a particular namespace.  There is
+// one PolicyBinding document per referenced Policy namespace
+type PolicyBinding struct {
+	unversioned.TypeMeta
+	// Standard object's metadata.
+	api.ObjectMeta
+
+	// LastModified is the last time that any part of the PolicyBinding was created, updated, or deleted
+	LastModified unversioned.Time
+
+	// PolicyRef is a reference to the Policy that contains all the Roles that this PolicyBinding's RoleBindings may reference
+	PolicyRef api.ObjectReference
+	// RoleBindings holds all the RoleBindings held by this PolicyBinding, mapped by RoleBinding.Name
+	RoleBindings map[string]*RoleBinding
+}
+
+// PolicyList is a collection of Policies
+type PolicyList struct {
+	unversioned.TypeMeta
+	// Standard object's metadata.
+	unversioned.ListMeta
+
+	// Items is a list of policies
+	Items []Policy
+}
+
+// PolicyBindingList is a collection of PolicyBindings
+type PolicyBindingList struct {
+	unversioned.TypeMeta
+	// Standard object's metadata.
+	unversioned.ListMeta
+
+	// Items is a list of policyBindings
+	Items []PolicyBinding
+}
+
+// RoleBindingList is a collection of RoleBindings
+type RoleBindingList struct {
+	unversioned.TypeMeta
+	// Standard object's metadata.
+	unversioned.ListMeta
+
+	// Items is a list of roleBindings
+	Items []RoleBinding
+}
+
+// RoleList is a collection of Roles
+type RoleList struct {
+	unversioned.TypeMeta
+	// Standard object's metadata.
+	unversioned.ListMeta
+
+	// Items is a list of roles
+	Items []Role
+}
+
+// ClusterRole is a logical grouping of PolicyRules that can be referenced as a unit by ClusterRoleBindings.
+type ClusterRole struct {
+	unversioned.TypeMeta
+	// Standard object's metadata.
+	api.ObjectMeta
+
+	// Rules holds all the PolicyRules for this ClusterRole
+	Rules []PolicyRule
+}
+
+// ClusterRoleBinding references a ClusterRole, but not contain it.  It can reference any ClusterRole in the same namespace or in the global namespace.
+// It adds who information via Users and Groups and namespace information by which namespace it exists in.  ClusterRoleBindings in a given
+// namespace only have effect in that namespace (excepting the master namespace which has power in all namespaces).
+type ClusterRoleBinding struct {
+	unversioned.TypeMeta
+	// Standard object's metadata.
+	api.ObjectMeta
+
+	// Subjects hold object references of to authorize with this rule
+	Subjects []api.ObjectReference
+
+	// RoleRef can only reference the current namespace and the global namespace
+	// If the ClusterRoleRef cannot be resolved, the Authorizer must return an error.
+	// Since Policy is a singleton, this is sufficient knowledge to locate a role
+	RoleRef api.ObjectReference
+}
+
+// ClusterPolicy is a object that holds all the ClusterRoles for a particular namespace.  There is at most
+// one ClusterPolicy document per namespace.
+type ClusterPolicy struct {
```

Comment:
Are you saying that there should be no cluster-wide policy, only per-namespace policy?  Or something else?